src/integrationTest/java/com/hivemq/helmcharts/CompatibilityHelmChartDeploymentIT.java

```diff
@@ -0,0 +1,58 @@
+package com.hivemq.helmcharts;
+
+import com.hivemq.client.mqtt.MqttGlobalPublishFilter;
+import com.hivemq.client.mqtt.datatypes.MqttQos;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5Client;
+import com.hivemq.helmcharts.util.OperatorHelmChartContainer;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test that the chart is deployed successfully on specific kubernetes cluster versions
+ */
+@SuppressWarnings("DuplicatedCode")
+@Testcontainers
+public class CompatibilityHelmChartDeploymentIT {
+
+    @Timeout(value = 20, unit = TimeUnit.MINUTES)
+    @ParameterizedTest
+    @ValueSource(strings = {"v1.20.15-k3s1", "v1.21.10-k3s1", "v1.22.7-k3s1", "v1.23.4-k3s1"})
+    public void withHelmLocalVersionDeployment_mqttMessagePublishedReceived(final @NotNull String version) throws Exception {
+        try (final var container = new OperatorHelmChartContainer(version, "k3s.dockerfile",
+                "values/customTestValues.yaml")) {
+            container.withLocalImages();
+            container.start();
+            final var client = Mqtt5Client.builder()
+                    .automaticReconnectWithDefaultConfig()
```

Comment:
is reconnect really necessary here?

Reply:
Yes, otherwise tests will not, work, it seems an issue between when the operator says the broker is ready and actually receives connections